src/LanguageServer/Impl/Implementation/Server.WorkspaceSymbols.cs

```diff
@@ -78,10 +78,9 @@ await GetModuleVariablesAsync(entry as ProjectEntry, opts, @params.query, 50, ca
             var all = breadthFirst.SelectMany(c => analysis.GetAllAvailableMembersFromScope(c, opts));
             var result = all
                 .Where(m => {
-                    if (m.Values.Any(v => v.DeclaringModule == entry || v.Locations.Any(l => l.DocumentUri == entry.DocumentUri))) {
-                        if (string.IsNullOrEmpty(prefix) || m.Name.StartsWithOrdinal(prefix, ignoreCase: true)) {
-                            return true;
-                        }
+                    if (m.Values.Any(v => v.DeclaringModule == entry || 
+                        v.Locations.MaybeEnumerate().Any(l => l.DocumentUri == entry.DocumentUri))) {
```

Comment:
Unless I'm misreading it, the stack trace in #378 seems to indicate that the exception comes from inside the function given to the second `Any`. Are we sure `DocumentUri` is never null?

Reply:
It appears to fail on `GetModuleVariables>b__4(ILocationInfo l)` which may happen if `v.Locations` is null - which is possible. `MaybeEnumerate` handles null to the empty enumeration.  Now, `DocumentUri` does can be null but that's OK.

Reply:
Sorry, I misspoke. I meant that `l` is null, i.e. an element inside of `Locations`. I read the exception as "we tried to access `l.DocumentUri` and that failed because `l` was null. If `v.Locations` were null, the crash can't have come from inside the function, because `Any` would have failed before actually calling it.

Reply:
Interesting point. I do see other places do `Locations.MaybeEnumerate()` which means `Locations` can be null. However, I don't see anything checking the actual location in the collection for nulls.

Reply:
For what it's worth, when I was trying to debug this myself, I removed all of the "wait for analysis" from the top of these calls, and did end up with exceptions with a slightly different stack trace which indicated that `Locations` was null. I added a load of debug assertions to this function for every one of these checks, but couldn't make them trigger on my desktop (and did not get around to trying it out on my laptop where I was able to reproduce the issue).